lib/collection/src/lib.rs

```diff
@@ -123,11 +123,17 @@ impl Collection {
         let mut shards: HashMap<ShardId, Shard> = HashMap::new();
         for shard_id in 0..config.params.shard_number.get() {
             let shard_path = shard_path(path, shard_id);
-            let shard = create_dir_all(&shard_path)
-                .map_err(|err| CollectionError::ServiceError {
+            let shard = tokio::fs::create_dir_all(&shard_path).await.map_err(|err| {
```

Comment:
Thanks to this change, we will be able to create the shards in parallel as well.
Given that we will also initialize remote shards it could be a nice to have.

Not asking you to do it, I could take care of it myself when adding the remote shards.